core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/clazz/list/ClazzListViewModel.kt

```diff
@@ -161,17 +165,23 @@ class ClazzListViewModel(
     }
 
     override fun onClickAdd() {
-        navigateToCreateNew(ClazzEditViewModel.DEST_NAME)
+        navigateToCreateNew(
+            ClazzEditViewModel.DEST_NAME,
+            extraArgs = mapOf(UstadView.ARG_CAN_ADD_COURSE to _uiState.value.canAddNewCourse.toString())
+        )
     }
-
     fun onClickJoinExistingClazz() {
-        navController.navigate(JoinWithCodeView.VIEW_NAME, mapOf(
-            UstadView.ARG_CODE_TABLE to Clazz.TABLE_ID.toString()
-        ))
+        navController.navigate(
+            JoinWithCodeView.VIEW_NAME, mapOf(
+                UstadView.ARG_CODE_TABLE to Clazz.TABLE_ID.toString()))
     }
-
     fun onClickEntry(entry: Clazz) {
-        navigateOnItemClicked(ClazzDetailViewModel.DEST_NAME, entry.clazzUid, entry)
+        navigateOnItemClicked(
+            ClazzDetailViewModel.DEST_NAME,
+            entry.clazzUid,
+            entry,
+            mapOf(UstadView.ARG_CAN_ADD_COURSE to _uiState.value.canAddNewCourse.toString())
```

Comment:
As mentioned above - use of UstadView.ARG_CAN_ADD_COURSE must be removed **entirely**.